### PR TITLE
Add support for Barbican Secret ACLs 

### PR DESCRIPTION
**Pull request by kowsalyapalaniappan**
Added support to update, delete and get acl details of secrets and containers. issue https://github.com/fog/fog-openstack/issues/278

### CODE REVIEW

File: lib/fog/key_manager/openstack.rb [hunk 1]
```diff
@@ -41,6 +42,12 @@ class OpenStack < Fog::Service
       request :list_containers
       request :delete_container
 
+      #acl
+      request :get_acl
```
Comment (Carthaca) [inline lib/fog/key_manager/openstack.rb]: please list the available requests (e.g. `get_secret_acl`) to make them discoverable, not the file names

File: lib/fog/key_manager/openstack/models/acl.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+require 'fog/openstack/models/model'
+require 'uri'
```
Comment (Carthaca) [inline lib/fog/key_manager/openstack/models/acl.rb]: unnecessary require

File: lib/fog/key_manager/openstack/models/acl.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+module Fog
+  module KeyManager
+    class OpenStack
+
+      class Acl < Fog::OpenStack::Model
```
Comment (Carthaca) [inline lib/fog/key_manager/openstack/models/acl.rb]: I suggest using upper case `ACL` here

File: lib/fog/key_manager/openstack.rb (hunk 1 above)
Comment (kowsalyapalaniappan) [inline lib/fog/key_manager/openstack.rb]: made this change in latest commit.

File: lib/fog/key_manager/openstack/requests/get_acl.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+      end
+
+      class Mock
+
+        def get_secret_acl(_uuid)
+          response = Excon::Response.new
+          response.status = 200
+          response.body = {
+              "data" => {
```
Comment (Carthaca) [inline lib/fog/key_manager/openstack/requests/get_acl.rb]: Thanks for adding some mocks :) Would you please format them a bit more? And I believe the keys `data` and `body` are wrong here, i.e. not needed to wrap the response's body

Comment (kowsalyapalaniappan) [inline lib/fog/key_manager/openstack/requests/get_acl.rb]: i splited the files and updated the mocks.

**Comment by Carthaca** [comment]
thanks, I just added the mock class and fixed some style

**Comment by kowsalyapalaniappan** [comment]
Thank you

**Pull request merged**